modules/apps/object/object-service/src/main/java/com/liferay/object/service/impl/ObjectFieldLocalServiceImpl.java

```diff
@@ -611,6 +611,25 @@ public ObjectField updateCustomObjectField(
 		return newObjectField;
 	}
 
+	@Indexable(type = IndexableType.REINDEX)
+	@Override
+	public ObjectField updateDBTableName(String dbTableName, long objectFieldId)
+		throws PortalException {
+
+		ObjectField objectField = getObjectField(objectFieldId);
+
+		return updateCustomObjectField(
+			objectField.getExternalReferenceCode(),
+			objectField.getObjectFieldId(),
+			objectField.getListTypeDefinitionId(),
+			objectField.getBusinessType(), dbTableName, objectField.getDBType(),
+			objectField.getDefaultValue(), objectField.isIndexed(),
+			objectField.isIndexedAsKeyword(),
+			objectField.getIndexedLanguageId(), objectField.getLabelMap(),
+			objectField.getName(), objectField.isRequired(),
+			objectField.isState(), objectField.getObjectFieldSettings());
```

Comment:
@seltong I would simplify by doing: 

```
if (Validator.isNotNull(objectField.getDBTableName()) {
   return objectField;
}

ObjectDefinition objectDefinition =
	_objectDefinitionPersistence.findByPrimaryKey(
		objectField.getObjectDefinitionId());

if (objectDefinition.isApproved()) {
	objectField.setDBTableName(objectDefinition.getExtensionDBTableName());
}
else {
	objectField.setDBTableName(objectDefinition.getDBTableName());
}

return objectFieldPersistence.update(objectField);
```

could you confirm if this works?

Reply:
@seltong I now see a problem with this logic... we have no way of knowing if the field was created before or after the object was published. 
In which scenario do we need to update the `DBTableName`?

Reply:
We need to update the DBTableName when the ObjectDefinition is not published. After published, I believe we can't update the dbTableName, right? @gabrielwas

Reply:
This kind of situation just happens when the object is imported. So, it'll be always before the publishing